### DISCUSSION TITLE
ESP8266 ADC input

### DISCUSSION THREAD

**Discussion by mmliam** [micropython/micropython#11232] [General]
The spec for the ESP8266 states that 1v at the ADC input reads as 1024 (full 10bit scale). I've found the for the ESP8266 used in the ESP-07 module any voltage over 0.95v at the ADC input reads 1024; in other words 0.95v results in a full scale reading of 1024.
Has anyone else observed this discrepancy between the spec and actual?

**Comment by robert-hh** [comment]
The ADC of the ESP8266 is not overly precise, to be polite. You have to do your own calibration for the range. The lower end of the voltage range is fuzzy as well.

**Reply by mmliam** [reply]
Thanks robert, I thought that was probably the case. I've calibrated one unit, now I'll get to see just how sloppy it is when I code another unit - lol
